shell/middleware/authenticated.js

```diff
@@ -91,7 +88,7 @@ function setProduct(store, to, redirect) {
   (product && !store.getters['type-map/isProductRegistered'](product))) {
     store.dispatch('loadingError', new Error(store.getters['i18n/t']('nav.failWhale.productNotFound', { productNotFound: product }, true)));
 
-    return () => redirect(302, '/fail-whale');
+    return true;
```

Comment:
The redirect is handled by the loadingError method in the store. You can see the fail-whale and setting product still work.


https://github.com/rancher/dashboard/assets/55104481/278a7584-1e01-4002-a233-b58939e95c62